modules/afsmtp/afsmtp.c

```diff
@@ -225,12 +227,20 @@ afsmtp_dd_format_persist_name(LogThrDestDriver *d)
 /*
  * Worker thread
  */
+
 static void
-afsmtp_dd_msg_add_recipient(AFSMTPRecipient *rcpt, smtp_message_t message)
+afsmtp_dd_msg_add_recipient(AFSMTPRecipient *rcpt, gpointer user_data)
 {
-  gchar *hdr;
+  AFSMTPDriver *self = ((gpointer *)user_data)[0];
+  LogMessage *msg = ((gpointer *)user_data)[1];
+  smtp_message_t message = ((gpointer *)user_data)[2];
 
```

Comment:
Perhaps split the user data parsing and the actual code to perform the operation. The body should be extracted to a separate function with typesafe arts.